doc/engine.md

```diff
@@ -0,0 +1,215 @@
+* Is this gonna be useful to others as well?
+* Is the solution robust and not just a hack?
+
+## 2. Obligation to content creators
+
+Game developers and modders should be provided with a
+reasonable, stable working environment to create their content in.
+
+At no point should you pull the rug from under a developer by
+changing this environment in an incompatible way.
+
+New features should primarily serve the purpose of providing
+content creators with more freedom. Features that restrict creative
+freedom in any way are unacceptable.
+
+## 3. Audiovisual style and artwork
+
+While primarily hosting simple-looking, blocky voxel games, Minetest
+supports a wide range of artwork styles and accomodates creators
+of all skill levels, from beginner hobbyists to professionals.
+
+Making good art for use with Minetest should be no more difficult
+than in any other commonly available game engine.
+
+After picking an art style based on the available features, an artist should
+not be forced to deal with new features incompatible with their art.
+New visual features should always accomodate all visual styles, providing
+an opt-out for developers who have no use for them. This opt-out must
+come in the form of scripting API provisions.
+It is unacceptable for a creator or server admin to have to ask the user
+to configure the client in a certain way.
+
+A content creator's restrictions on presentation take precedence over
+personal preference. It should not be easy for a user to unknowingly
+configure their client in a way where they may get a sub-par experience,
+and possibly even blame the content creator for it.
+Understand that some types of artwork and some rendering features do not mix.
+A model made and textured for an unlit renderer is unlikely to look
+better with gouraud shading or shadows, it's more likely to look worse.
+
+Heed the words of one dissatisfied user: (Issue #10307, worth a read)
+
+> When people enable additional graphics features, they expect better graphics, not worse
+
+An implementation of a visual feature should aim for a balanced
+quality and performance profile that fits a general use case and works
+on a wide range of devices.
+
+Visual features that do not work on all supported devices may be regarded
+as low priority. (This does not apply to pure optimizations.)
+
+### 3.1. Client visual settings
+
+While the client may toggle various visual features for the sake
+of personal preference and performance, games should not be
+forced to utilize or anticipate those that significantly
+change the way content is being rendered.
+
+Examples of visually neutral settings:
+* Screen resolution
+* Maximum FPS
+* Vertical sync
+* Anti-aliasing
+* Texture filtering
+* Stereoscopy
+
+The above settings affect the image only in minor ways,
+and are safe to apply to just about any content.
+
+
+Examples of non-neutral features that should be API gated:
+* Direct lighting
+* Realtime shadows
+* Bump-mapping
```

Comment:
The old bumpmapping had a feature that would autogenerate normalmaps. That would clearly fail this requirement.
Suppose bumpmapping is readded but only took effect if correct named normal textures exist, would that count as "API gated"?

Reply:
I guess I didn't think of something like asset path rules but I'd consider that poor form as opposed to a `.normal_map` field in a tile definition. It's best if we avoid adding any more path quirks.

So Lua API is the preferred mechanism.